app/src/main/java/openfoodfacts/github/scrachx/openfood/fragments/AddProductIngredientsFragment.java

```diff
@@ -38,6 +73,21 @@ public View onCreateView(@NonNull LayoutInflater inflater, ViewGroup container,
         return view;
     }
 
+    @Override
+    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
+        super.onViewCreated(view, savedInstanceState);
+        Bundle b = getArguments();
+        if (b != null) {
+            Product product = (Product) b.getSerializable("product");
+            if (product != null) {
+                code = product.getCode();
+            }
+        } else {
+            Toast.makeText(activity, "Something went wrong", Toast.LENGTH_SHORT).show();
```

Comment:
Let's try to make the error messages more explicit. 
"Something went wrong when trying to add product ingredients."
Perhaps try giving possible causes, and a follow-up action ?